### PR TITLE
[iOS] Fixed crash displaying strings in CarouselView

### LINKED ISSUE #7789: [Bug] [iOS] CarouselView crash when displaying strings in 4.3-pre3
### Description

The following code crashes (SIGSEGV) on iOS in 4.3-pre3:

```
    <StackLayout Margin="10">
        <CarouselView>
            <CarouselView.ItemsSource>
                <x:Array Type="{x:Type x:String}">
                    <x:String>Baboon</x:String>
                    <x:String>Capuchin Monkey</x:String>
                    <x:String>Blue Monkey</x:String>
                    <x:String>Squirrel Monkey</x:String>
                    <x:String>Golden Lion Tamarin</x:String>
                    <x:String>Howler Monkey</x:String>
                    <x:String>Japanese Macaque</x:String>
                    <x:String>Mandrill</x:String>
                    <x:String>Proboscis Monkey</x:String>
                    <x:String>Red-shanked Douc</x:String>
                    <x:String>Gray-shanked Douc</x:String>
                    <x:String>Golden Snub-nosed Monkey</x:String>
                    <x:String>Black Snub-nosed Monkey</x:String>
                    <x:String>Tonkin Snub-nosed Monkey</x:String>
                    <x:String>Thomas's Langur</x:String>
                    <x:String>Purple-faced Langur</x:String>
                    <x:String>Gelada</x:String>
                </x:Array>
            </CarouselView.ItemsSource>
        </CarouselView>
    </StackLayout> 
```

Note this only occurs on iOS. Android works fine.

### Steps to Reproduce

1. Run the attached sample.
2. Browse to the HorizontalTextLayoutPage.

### Expected Behavior

Strings are displayed in a CarouselView.

### Actual Behavior

SIGSEGV:

=================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x105bc6ac5 - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : mono_dump_native_crash_info
	0x105bbb115 - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : mono_handle_native_crash
	0x105bcd2d1 - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : mono_sigsegv_signal_handler_debug
	0x115613b5d - /usr/lib/system/libsystem_platform.dylib : _sigtramp
	0x105c65af9 - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : mono_class_get_checked
	0x105bba1ce - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : mono_handle_exception_internal
	0x105bb92a9 - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : mono_handle_exception
	0x105b3d3af - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : mono_amd64_throw_exception
	0x1342bd5b0 - Unknown
	0x105def1e3 - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : _ZL17exception_handlerP11NSException
	0x112fbfbcb - /Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.2.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : __handleUncaughtException
	0x113ccee27 - /Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.2.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libobjc.A.dylib : _ZL15_objc_terminatev
	0x1149d7a2e - /Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.2.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libc++abi.dylib : _ZSt11__terminatePFvvE
	0x1149d74eb - /Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.2.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libc++abi.dylib : __cxa_get_exception_ptr
	0x1149d74b2 - /Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.2.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libc++abi.dylib : _ZN10__cxxabiv1L22exception_cleanup_funcE19_Unwind_Reason_CodeP17_Unwind_Exception
	0x113ccebfa - /Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.2.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libobjc.A.dylib : _ZL26_objc_exception_destructorPv
	0x105df0900 - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : xamarin_process_nsexception_using_mode
	0x105df07e7 - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : xamarin_process_nsexception
	0x105e00cc9 - /Users/davidbritch/Library/Developer/CoreSimulator/Devices/B44E7231-EE40-466A-BC26-3C1983455878/data/Containers/Bundle/Application/734294D5-D70D-4D9E-BDFB-F9D2B9A32157/CarouselViewDemos.iOS.app/CarouselViewDemos.iOS : xamarin_dyn_objc_msgSend
	0x1380bb75e - Unknown

=================================================================
	Basic Fault Address Reporting
=================================================================
Memory around native instruction pointer (0x105ca5b84):0x105ca5b74  47 20 5d c3 0f 1f 84 00 00 00 00 00 55 48 89 e5  G ].........UH..
0x105ca5b84  8b 47 20 a9 00 00 00 02 75 0c a9 00 00 00 10 75  .G .....u......u
0x105ca5b94  26 48 8b 07 5d c3 48 8d 3d 0e ee 20 00 48 8d 35  &H..].H.=.. .H.5
0x105ca5ba4  8b e9 20 00 48 8d 0d ff e9 20 00 ba 79 03 00 00  .. .H.... ..y...

=================================================================
	Managed Stacktrace:
=================================================================
	  at <unknown> <0xffffffff>
	  at ObjCRuntime.Messaging:IntPtr_objc_msgSend_IntPtr_IntPtr <0x0011d>
	  at UIKit.UICollectionView:DequeueReusableCell <0x002ea>
	  at UIKit.UICollectionView:DequeueReusableCell <0x000e3>
	  at Xamarin.Forms.Platform.iOS.ItemsViewController:GetCell <0x0010a>
	  at Xamarin.Forms.Platform.iOS.CarouselViewController:GetCell <0x000aa>
	  at <Module>:runtime_invoke_object__this___object_object <0x001d5>
	  at <unknown> <0xffffffff>
	  at UIKit.UIApplication:UIApplicationMain <0x00251>
	  at UIKit.UIApplication:Main <0x000b2>
	  at UIKit.UIApplication:Main <0x00132>
	  at CarouselViewDemos.iOS.Application:Main <0x00092>
	  at <Module>:runtime_invoke_void_object <0x00198>
=================================================================

### Basic Information

- Version with issue: 4.3.851321-pre3
- Last known good version: It didn't crash in 4.3-pre2, instead displaying nothing.
- IDE: VSMac 8.3.1
  - iOS:  13.0

### Reproduction Link

[CarouselViewDemos.zip](https://github.com/xamarin/Xamarin.Forms/files/3685340/CarouselViewDemos.zip)

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

 Fixed **crash** displaying strings in **CarouselView**. Example:

```
 <StackLayout Margin="10">
        <CarouselView>
            <CarouselView.ItemsSource>
                <x:Array Type="{x:Type x:String}">
                    <x:String>Baboon</x:String>
                    <x:String>Capuchin Monkey</x:String>
                    <x:String>Blue Monkey</x:String>
                </x:Array>
            </CarouselView.ItemsSource>
        </CarouselView>
</StackLayout> 
```
The crash was on `DequeueReusableCell` because with the CarouselView we was registering the cell used with template in the Carousel https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Platform.iOS/CollectionView/CarouselViewController.cs#L62. However, without template (Example: using strings) the cell expected was not registered: https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewController.cs#L253

### Issues Resolved ### 

- fixes #7789 

### API Changes ###

 None

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###
Launch Core Gallery and navigate to the Issue 7789. If the CarouselView is rendered (without any error), the test passes.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems [hunk 1]
```diff
@@ -1403,6 +1406,9 @@
     <Compile Update="$(MSBuildThisFileDirectory)Issue7357.xaml.cs">
       <DependentUpon>Issue7357.xaml</DependentUpon>
     </Compile>
+    <Compile Update="C:\Projects\Xamarin.Forms\Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\Issue7789.xaml.cs">
```
Comment (jfversluis) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems]: Hard path reference?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems [hunk 2]
```diff
@@ -1403,6 +1406,9 @@
...
+      <DependentUpon>Issue7789.xaml</DependentUpon>
+    </Compile>
     <Compile Update="C:\Users\hartez\Documents\Xamarin\Xamarin.Forms\Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\Issue7519Xaml.xaml.cs">
```
Comment (jfversluis) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems]: While this is not part of your change, also: Hard path reference?

**Comment by jfversluis** [comment]
In addition, this is marked regression, does it need to go into 4.2?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems (hunk 2 above)
Comment (jsuarezruiz) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems]: Oops, I hadn't realized this. Thanks!.

**Comment by jsuarezruiz** [comment]
@jfversluis The issue is on 4.3-pre3 and was not crashing on 4.3-pre2, for that the target is 4.3.0.

**Comment by jfversluis** [comment]
@jsuarezruiz seems there's a conflict can you solve it/rebase it? After that, feel free to merge :)

**Comment by jsuarezruiz** [comment]
Fixed the issue and merged!.

**Pull request merged**